packages/shared/src/client/client.ts

```diff
@@ -1,7 +1,6 @@
-import { Metadata } from '../types';
 import { ProviderMetadata } from '../provider/provider';
 
-export interface ClientMetadata extends Metadata {
```

Comment:
`Metadata` wasn't doing much here since it was an empty interface.

Reply:
Maybe we should do `extends Readonly<Metadata>` here as well, so we can add arbitrary stuff in client's just the same as providers? I know it wasn't doing anything before, but with your changes now it actually adds functionality.

If you agree, we can make this change and merge.

Reply:
I'm not sure how custom metadata would be set on the client.